src/redis.c

```diff
@@ -319,10 +317,12 @@ static int redis_handle_query(redisContext *rh, redis_node_t *rn,
     return -1;
   }
 
-  if ((rr = redisCommand(rh, "SELECT %d", rq->db)) == NULL) {
-    WARNING("redis plugin: unable to switch to db `%d' on node `%s'.", rq->db,
-            rn->name);
-    return -1;
+  if (rq->database > 0) {
```

Comment:
Ouch, sorry for new and new comments, but this rule (`database > 0`) becomes wrong when we put `Database` to `Query` level.
If you set `Database 0` in some `Query` then database index from previous `Query` will be used, what will be unexpected and wrong behaviour.

Everything else looks good, with one minor note: Maybe it will be good to put new option in examples as commented, to emphasize it is not mandatory. On your selection.